File: priority scheduling.py
print("Enter the number of processess: ")
n = int(input())
processes = []
for i in range(0, n):
    processes.insert(i, i + 1)

print("Enter the burst time of the processes: ")
bt = list(map(int, input().split()))

print("Enter the priority of the processes: ")
priority = list(map(int, input().split()))
finish = []
wt = []

for i in range(0, len(priority) - 1):
    for j in range(0, len(priority) - i - 1):
        if (priority[j] > priority[j + 1]):
            swap = priority[j]
            priority[j] = priority[j + 1]
            priority[j + 1] = swap

            swap = bt[j]
            bt[j] = bt[j + 1]
            bt[j + 1] = swap

            swap = processes[j]
            processes[j] = processes[j + 1]
            processes[j + 1] = swap

wt.insert(0, 0)
finish.insert(0, bt[0])
for i in range(1, len(processes)):
    wt.insert(i, wt[i - 1] + bt[i - 1])
    finish.insert(i, wt[i] + bt[i])
avgwt = 0
for i in range(0, len(processes)):
    avgwt = avgwt + wt[i]
avgwt = float(avgwt) / n
print(wt[0], end=" ")
for i in range(0,n):
 print(" "+"P"+str(i+1)+" "+str(finish[i]), end=" ")
print("\n")
print("Average Waiting time is: " + str(avgwt))

File: priority scheduling.cpp
#include<stdio.h>
#define max 20
int main()
{
    int i,j,n,bt[max],p[max],wt[max],tat[max],pr[max],total=0,pos,temp;
    float avg_wt;
    printf("Number of Process:");
    scanf("%d",&n);

    printf("\nEnter Burst Time and Priority\n");
    for(i=0;i<n;i++)
    {
        printf("Burst Time for P[%d]:", i+1);
        scanf("%d",&bt[i]);
    }
    printf("\n");
    for(i=0;i<n;i++){

        printf("Priority for P[%d]:",i+1);
        scanf("%d",&pr[i]);
        p[i]=i+1;
    }

    for(i=0;i<n;i++)
    {
        pos=i;
        for(j=i+1;j<n;j++)
        {
            if(pr[j]<pr[pos])
                pos=j;
        }

        temp=pr[i];
        pr[i]=pr[pos];
        pr[pos]=temp;

        temp=bt[i];
        bt[i]=bt[pos];
        bt[pos]=temp;

        temp=p[i];
        p[i]=p[pos];
        p[pos]=temp;
    }

    wt[0]=0;

    for(i=1;i<n;i++)
    {
        wt[i]=0;
        for(j=0;j<i;j++)
            wt[i]+=bt[j];

        total+=wt[i];
    }

    avg_wt=total/n;
    total=0;

    printf("\nGantt Chart : \n");

    printf("0 ");
    for(i=0;i<n;i++)
    {
        wt[i]=bt[i]+wt[i];
        printf("P%d %d ",p[i],wt[i]);
    }

    printf("\n\nAverage Waiting Time=%f",avg_wt);

	return 0;
}

